#! /bin/sh /usr/share/dpatch/dpatch-run
## fix-multipart.sl.dpatch by Brian Murray <brian@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: The usage of the return value of string_match in multipart.sl is
## DP: wrong. The return value is the position from the start of the string
## DP: and not from the start of the match. Hence, the calculation of the
## DP: new start is wrong.

@DPATCH@

diff --git a/macros/multipart.sl b/macros/multipart.sl
--- a/macros/multipart.sl
+++ b/macros/multipart.sl
@@ -40,7 +40,7 @@ define subject_compare_hook ( subject1, subject2)
 
    while (tmp > 0)
    {
-      pos1 = pos1 + 4 + tmp;
+      pos1 = tmp;
       tmp = string_match(subject1, pattern, pos1 + 5);
    }
 
@@ -48,7 +48,7 @@ define subject_compare_hook ( subject1, subject2)
    tmp = string_match(subject2, pattern, pos2 + 5);
    while (tmp > 0)
    {
-      pos2 = pos2 + 4 + tmp;
+      pos2 = tmp;
       tmp = string_match(subject2, pattern, pos2 + 5);
    }
         
